src/coverlet.collector/DataCollection/CoverletSettingsParser.cs

```diff
@@ -205,5 +205,15 @@ private bool ParseIncludeTestAssembly(XmlElement configurationElement)
             bool.TryParse(includeTestAssemblyElement?.InnerText, out bool includeTestAssembly);
             return includeTestAssembly;
         }
+
+        /// <summary>
+        /// Splits a comma separated elements into an array
+        /// </summary>
+        /// <param name="element">The element to split</param>
+        /// <returns>An array of the values in the element</returns>
+        private string[] SplitElement(XmlElement element)
+        {
+            return element?.InnerText?.Split(',')?.Select(value => value.Trim())?.ToArray();
```

Comment:
Maybe `return element?.InnerText?.Split(',', StringSplitOptions.RemoveEmptyEntries).Select(value => value.Trim()).ToArray();` why the null check after `InnerText` one?

Reply:
I've updated it and added a coalesce operator at the end to return an empty array instead of null. The extra null checks weren't needed so they are gone.

Reply:
> I've updated it and added a coalesce operator at the end to return an empty array instead of null. The extra null checks weren't needed so they are gone.

I prefer to keep old semantic and return null, I'd like to avoid surprise in case of expected null of callers. Please remove the coalesce at the end, the rest is ok to me.

Reply:
It's also a shame, maybe in future we'll update with elegant one https://github.com/dotnet/corefx/issues/41487#issuecomment-573576875

Reply:
Coalesce has been removed